common/persistence/sql/sqlTaskManager.go

```diff
@@ -74,39 +73,40 @@ func (m *sqlTaskManager) LeaseTaskQueue(request *persistence.LeaseTaskQueueReque
 		return nil, serviceerror.NewInternal(err.Error())
 	}
 	tqId, tqHash := m.taskQueueIdAndHash(nidBytes, request.TaskQueue, request.TaskType)
+	var row sqlplugin.TaskQueuesRow
 	rows, err := m.db.SelectFromTaskQueues(sqlplugin.TaskQueuesFilter{RangeHash: tqHash, TaskQueueID: tqId})
-	if err != nil {
-		if err == sql.ErrNoRows {
-			namespaceID := request.NamespaceID
-			tqInfo := &persistencespb.TaskQueueInfo{
-				NamespaceId:    namespaceID,
-				Name:           request.TaskQueue,
-				TaskType:       request.TaskType,
-				AckLevel:       ackLevel,
-				Kind:           request.TaskQueueKind,
-				ExpiryTime:     nil,
-				LastUpdateTime: timestamp.TimeNowPtrUtc(),
-			}
-			blob, err := serialization.TaskQueueInfoToBlob(tqInfo)
-			if err != nil {
-				return nil, err
-			}
-			row := sqlplugin.TaskQueuesRow{
-				RangeHash:    tqHash,
-				TaskQueueID:  tqId,
-				Data:         blob.Data,
-				DataEncoding: blob.EncodingType.String(),
-			}
-			rows = []sqlplugin.TaskQueuesRow{row}
-			if _, err := m.db.InsertIntoTaskQueues(&row); err != nil {
-				return nil, serviceerror.NewInternal(fmt.Sprintf("LeaseTaskQueue operation failed. Failed to make task queue %v of type %v. Error: %v", request.TaskQueue, request.TaskType, err))
-			}
-		} else {
-			return nil, serviceerror.NewInternal(fmt.Sprintf("LeaseTaskQueue operation failed. Failed to check if task queue existed. Error: %v", err))
-		}
-	}
-
-	row := rows[0]
+	switch err {
+	case nil:
+		row = rows[0]
+	case sql.ErrNoRows:
+		namespaceID := request.NamespaceID
+		tqInfo := &persistencespb.TaskQueueInfo{
+			NamespaceId:    namespaceID,
+			Name:           request.TaskQueue,
+			TaskType:       request.TaskType,
+			AckLevel:       ackLevel,
+			Kind:           request.TaskQueueKind,
+			ExpiryTime:     nil,
+			LastUpdateTime: timestamp.TimeNowPtrUtc(),
+		}
+		blob, err := serialization.TaskQueueInfoToBlob(tqInfo)
+		if err != nil {
+			return nil, err
+		}
+		row = sqlplugin.TaskQueuesRow{
+			RangeHash:    tqHash,
+			TaskQueueID:  tqId,
+			RangeID:      0,
+			Data:         blob.Data,
+			DataEncoding: blob.EncodingType.String(),
+		}
+		if _, err := m.db.InsertIntoTaskQueues(&row); err != nil {
+			return nil, serviceerror.NewInternal(fmt.Sprintf("LeaseTaskQueue operation failed. Failed to make task queue %v of type %v. Error: %v", request.TaskQueue, request.TaskType, err))
+		}
+	default:
+		return nil, serviceerror.NewInternal(fmt.Sprintf("LeaseTaskQueue operation failed. Failed to check if task queue existed. Error: %v", err))
```

Comment:
[Nit] since we're here, might as well also dump the TaskQueue name and TaskQueue type...